### ISSUE TITLE
Potential Issue with Styles in LightboxJS Library

### ISSUE THREAD

**Issue by ObieSilva** [silvia-odwyer/lightbox.js#61]
**Description:** I recently experienced two issues with the LightboxJS library that affected both the visibility of the Lightbox content and the display of thumbnails:

- **Opacity Issue:** Certain elements within the Lightbox were styled with opacity: 0, causing the content to be invisible. This was occurring for elements with the classes ._PH6f8 and ._3pfLz.
- **Thumbnail Positioning:** The thumbnails were not displaying correctly, and adjustments were needed to make them visible again.

**Background:**
- **Library Version:** 1.1.3
- **Environment:** NextJS, Tailwind, TypeScript
- **Context:** This issue did not occur previously, and the Lightbox was functioning as expected. The opacity property being set to 0 seems to come directly from the LightboxJS library, and I have verified that my implementation and imports are correct based on the documentation.

**Steps Taken to Troubleshoot:**
- Removed all custom and global CSS, including Tailwind CSS, to ensure there were no conflicts.
- Updated the LightboxJS library to the latest version and double-checked the documentation to ensure proper implementation.
- Inspected the element using browser developer tools, which showed that the inline opacity: 0 style was causing the content to remain hidden.

Temporarily fixed the issue by overriding the opacity with custom CSS:
`._PH6f8, ._3pfLz {
  opacity: 1 !important; /* Force visibility */
}

._-wN_b {
  bottom: 24% !important; /* Adjust thumbnail positioning */
}`

Request: Could you please look into this issue and let me know if there are any configuration changes or updates required on my part? Alternatively, if this is a bug, an update or workaround from your end would be greatly appreciated.

Please let me know if you need any additional information, such as a code snippet or a reproduction of the issue.

**Comment by silvia-odwyer** [comment]
@ObieSilva Hi Obie,
Thanks for opening this issue. I'm just wondering if this error occurs with the latest version of the library?
Also, could you provide the JSX you're using to display the lightbox? This will help me to debug the issue. 
Thanks very much!

**Comment by ObieSilva** [comment]
@silvia-odwyer - Apologies for the late reply.

I'll be honest—the last time I checked, the application I’m building was working well with Next.js version 14.2.14. However, when I reverted back to that version for testing purposes, everything worked fine on desktop and tablet but not on mobile. The mini thumbnails didn’t display, and the left and right arrows were misaligned. Please see screenshots ...

![desktop](https://github.com/user-attachments/assets/a18cb14b-2eae-4866-b288-fe0ca8405b26)
![mobile](https://github.com/user-attachments/assets/14bac843-8387-4185-ae12-1575277581e9)
![tablet](https://github.com/user-attachments/assets/c624da53-8edc-4030-84e1-4ab7e830431a)

Here is the code I’m using for the lightbox where the issue persists. The current version of Next.js is 15.0.3.

`import { useEffect } from "react";
import { SlideshowLightbox, initLightboxJS } from "lightbox.js-react";
import "lightbox.js-react/dist/index.css";
import Image from "next/image";

interface Photo {
  src: string;
  alt: string;
}

interface CarGalleryProps {
  photos: Photo[];
  lightboxId?: string;
}

/**
 * CarGallery is a gallery component that displays a large image and uses LightboxJS for slideshow functionality.
 * The LightboxJS library is initialized locally within this component to avoid globally loading unnecessary libraries.
 *
 * @param {Photo[]} photos - Array of photo objects with src and alt properties.
 * @param {string} [lightboxId='lightbox1'] - Optional Lightbox identifier for managing instances.
 * @returns {JSX.Element} The rendered lightbox gallery component.
 */
const CarGallery: React.FC<CarGalleryProps> = ({
  photos,
  lightboxId = "lightbox1",
}) => {
  useEffect(() => {
    const licenseKey = process.env.NEXT_PUBLIC_LIGHTBOX_LICENSE_KEY;
    const planType = process.env.NEXT_PUBLIC_LIGHTBOX_PLAN_TYPE;

    if (licenseKey && planType) {
      initLightboxJS(licenseKey, planType); // Initialize LightboxJS locally for this component
    }
  }, []);

  return (
    <SlideshowLightbox
      lightboxIdentifier={lightboxId}
      framework="next"
      images={photos}
      showThumbnails={true}
      className="relative h-64 md:h-[28rem] lg:h-[31.9rem] xl:h-[30.3rem]"
    >
      {photos.length > 0 && (
        <Image
          src={photos[0].src}
          alt={photos[0].alt}
          fill
          data-lightboxjs={lightboxId}
          quality={80}
          className="object-cover rounded-lg drop-shadow-2xl"
          priority
        />
      )}
    </SlideshowLightbox>
  );
};

export default CarGallery;`

**Comment by silvia-odwyer** [comment]
@ObieSilva Hi Obie,
I believe the issue may be related to the version of the animation library I was using. I've updated this to the latest version, could you check if it resolves the issue?
If not, I've also added a new prop called `disableAnim`, which disables any animations from the library when set to `true`. The lightbox should appear then.

Also, when importing the CarGallery component and using Next.js 15, be sure to import the component dynamically with `ssr `set to `false`, like so:
```jsx
const CarGallery= dynamic(() => import('./path-to-cargallery-component'), { ssr: false })
```
While debugging this issue, I noticed an error when I didn't dynamically import the component that displayed the lightbox.
Let me know how this goes, thanks!

**Comment by ObieSilva** [comment]
@silvia-odwyer - I made the changes you mentioned, however, now I get a different error ... It seems as the index.css is missing?

![image](https://github.com/user-attachments/assets/e846ffcb-7cc8-495c-819f-59b58e4b79fe)

I've cleaned the cache on npm, reinstalled the package - Could this be something on my end? Nothing else has changed outside of just making the changes mentioned and updating the library. I will revert the changes and try again later as I need to head out but just wanted to point it out.

**Comment by silvia-odwyer** [comment]
@ObieSilva Yes, the index.css file is no longer required. You can remove that line, and the styles should be automatically applied whenever you use the components. ✅

**Comment by ObieSilva** [comment]
@silvia-odwyer - Perfect, functionality is working properly. The only issue I still see is in mobile but is due to CSS so I can quickly write a fix for this on my end :)

![image](https://github.com/user-attachments/assets/4934bf59-0550-459b-9fb4-7ad7a4046d07)

You have been amazing to fix this quickly and thank you for creating this tool for us.

**Comment by dandc11** [comment]
With version `1.4.4` of the library, I'm getting cut off displays on both desktop and mobile. I'm using Next.js 14.2.7, React 18.3.1.

On desktop, this is the lightbox display without any style adjustments

<img width="1311" alt="Image" src="https://github.com/user-attachments/assets/8c58b7a7-00c8-4384-af4d-f9de4d4454c1" />

On mobile, here's the default display of the same image, which is even more cut off. 

<img width="328" alt="Image" src="https://github.com/user-attachments/assets/c1c12294-48f4-41c4-9e67-3dea68b44a4a" />

My config: 

<img width="404" alt="Image" src="https://github.com/user-attachments/assets/bf02306b-e428-4d3a-9ff8-077aa20f948b" />

**Comment by silvia-odwyer** [comment]
@dandc11 Hi Dan,
I've just released a new version of the library which should fix the layout issue. However if it still persist, just let me know and I'll take a look. Thanks!

**Comment by dandc11** [comment]
Hi @silvia-odwyer ,

That fixed the centering issue for the featured image, but the width and height settings the library is applying are distorting the image on desktop. 

![Image](https://github.com/user-attachments/assets/1dfabcb7-a98b-48f4-864e-92a938fcc596)

![Image](https://github.com/user-attachments/assets/4c64a4d6-aa84-49a2-81c5-52c077b5df88)

Additionally, the captions are being covered on mobile when thumbnails are shown. 

![Image](https://github.com/user-attachments/assets/948ce2a8-e9ff-417e-8767-45d9d0c37078)

**Comment by silvia-odwyer** [comment]
@dandc11 Hi Dan,
I've just updated the library to incorporate fixes for the layout issues you mentioned. I updated the image `object-fit` property so that it uses `contain` instead. 
Can you let me know if it fixes the issues mentioned? Thanks!
